src/storage-types/src/sources/mysql.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Types related to mysql sources
+
+use mz_proto::{IntoRustIfSome, RustType, TryFromProtoError};
+use mz_repr::{ColumnType, GlobalId, RelationDesc, ScalarType};
+use once_cell::sync::Lazy;
+use proptest::prelude::{any, Arbitrary, BoxedStrategy, Strategy};
+use proptest_derive::Arbitrary;
+use serde::{Deserialize, Serialize};
+
+use crate::connections::inline::{
+    ConnectionAccess, ConnectionResolver, InlinedConnection, IntoInlineConnection,
+    ReferencedConnection,
+};
+use crate::sources::SourceConnection;
+
+include!(concat!(
+    env!("OUT_DIR"),
+    "/mz_storage_types.sources.mysql.rs"
+));
+
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct MySqlSourceConnection<C: ConnectionAccess = InlinedConnection> {
+    pub connection_id: GlobalId,
+    pub connection: C::MySql,
+    pub details: MySqlSourceDetails,
+}
+
+impl<R: ConnectionResolver> IntoInlineConnection<MySqlSourceConnection, R>
+    for MySqlSourceConnection<ReferencedConnection>
+{
+    fn into_inline_connection(self, r: R) -> MySqlSourceConnection {
+        let MySqlSourceConnection {
+            connection_id,
+            connection,
+            details,
+        } = self;
+
+        MySqlSourceConnection {
+            connection_id,
+            connection: r.resolve_connection(connection).unwrap_mysql(),
+            details,
+        }
+    }
+}
+
+impl<C: ConnectionAccess> Arbitrary for MySqlSourceConnection<C> {
+    type Strategy = BoxedStrategy<Self>;
+    type Parameters = ();
+
+    fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
+        (
+            any::<C::MySql>(),
+            any::<GlobalId>(),
+            any::<MySqlSourceDetails>(),
+        )
+            .prop_map(|(connection, connection_id, details)| Self {
+                connection,
+                connection_id,
+                details,
+            })
+            .boxed()
+    }
+}
+
+pub static MYSQL_PROGRESS_DESC: Lazy<RelationDesc> = Lazy::new(|| {
+    RelationDesc::empty()
+        // TODO(petrosagg): UUIDs in ranges are not supported and additionally the
+        // Partitioned timestamp type forces you to deal with open/closed ranges which are
```

Comment:
As of https://github.com/MaterializeInc/materialize/pull/24233 this is now fixed so we could set this to be:

```
RelationDesc::empty()
    .with_column("source_id_lower", ScalarType::Uuid.nullable(false))
    .with_column("source_id_upper", ScalarType::Uuid.nullable(false))
    .with_column("transaction_id", ScalarType::UInt64.nullable(true)) // <-- notice this changed too
```